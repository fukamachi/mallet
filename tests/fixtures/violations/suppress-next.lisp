;; Test file for suppress-next declarations

(defpackage #:test-suppress-next
  (:use #:cl))
(in-package #:test-suppress-next)

;; This function has if-without-else but is suppressed
#+mallet
(declaim (mallet:suppress-next :if-without-else))
(defun foo (x)
  (if x
      (print "yes")))  ; No else clause - but suppressed!

;; This should trigger if-without-else violation (no suppression)
(defun bar (x)
  (if x
      (print "also yes")))  ; No else clause - should be flagged
