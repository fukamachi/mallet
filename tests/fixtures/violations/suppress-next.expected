# Expected violations for suppress-next.lisp
# Format: line:column rule-name severity
# Only the second function (bar) should have a violation
# The first function (foo) has suppress-next, so its violation is suppressed

16:0 if-without-else convention
